src/lib/CLValue2/List.test.ts

```diff
@@ -0,0 +1,114 @@
+import { expect } from 'chai';
+import { List } from './List';
+import { Bool, BoolType } from './Bool';
+
+describe('CLValue List implementation', () => {
+  it('Bool should return proper clType', () => {
+    const myBool = new Bool(false);
+    const clType = myBool.clType();
+    expect(clType.toString()).to.be.eq("Bool");
+  });
+
+  it('List should return proper clType', () => {
+    const myBool = new Bool(false);
+    const myList = new List<Bool>([myBool]);
+
+    expect(myList.clType().toString()).to.be.eq("List (Bool)");
+  });
+
+  it('Should be able to create List with proper values - correct by construction', () => {
+    const myList = new List([new Bool(true), new Bool(false)]);
+
+    expect(myList).to.be.an.instanceof(List);
+  });
+
+  it('Should throw an error when list is not correct by construction', () => {
+    const badFn = () => new List([new Bool(true), new List([new Bool(false)])]);
+
+    expect(badFn).to.throw("Invalid data provided.");
+  });
+
+  it('Should throw an error when list is not correct by construction', () => {
+    const badFn = () => new List([1, 2, 3]);
+
+    expect(badFn).to.throw("Invalid data type(s) provided.");
+  });
+
+  it('Should be able to return proper values by calling .value() on List', () => {
+    const myBool = new Bool(false);
+    const myList = new List<Bool>([myBool]);
```

Comment:
`<Bool>` is not required here. In general we shouldn't be using it like this in tests, as it's not available in JS.